### PR TITLE
Fix jdk detection

### PR DESCRIPTION
**Pull request by Solant**
- Fixed JDK detection for case when `java` and `javac` are in different locations.
- Added test for JDK detection
- Fixed test for unique paths of java home locations (`/usr/lib/jvm/default` and `/usr/lib/jvm/default-runtime/jre` were considered identical)

### CODE REVIEW

File: ts/test/index.ts [hunk 1]
```diff
@@ -71,8 +80,6 @@ describe(`CLI Tests`, () => {
     const output = spawnSync('node', [LJH_BIN]);
     const lines = output.stdout.toString().trim().split("\n");
     assertEqual(lines.length, javaHomes.length + 1, `Should print all found JAVA_HOMES`);
-    const firstJavaHome = javaHomes[0];
-    const firstJavaHomeLines = lines.filter((l) => l.indexOf(firstJavaHome.path) !== -1);
-    assertEqual(firstJavaHomeLines.length, 1, `There should be one line of output for each JAVA_HOME found.`);
+    assertEqual(lines.length, new Set(lines).size, `There should be one line of output for each JAVA_HOME found.`);
```
Comment (jvilk) [inline ts/test/index.ts]: Much cleaner. I wasn't thinking very well when I wrote the original test code!

**Comment by jvilk** [comment]
Ah, excellent edge case! Thanks for documenting your code and adding tests. I'll merge in and update the NPM package.
